analyzers/src/SonarAnalyzer.CSharp/Helpers/CSharpSymbolUsageCollector.cs

```diff
@@ -298,34 +273,34 @@ private void StorePropertyAccess(IPropertySymbol propertySymbol, AccessorAccess
         private AccessorAccess EvaluatePropertyAccesses(ExpressionSyntax node)
         {
             var topmostSyntax = GetTopmostSyntaxWithTheSameSymbol(node);
-
             if (topmostSyntax.Parent is AssignmentExpressionSyntax assignmentExpression)
             {
                 if (assignmentExpression.IsKind(SyntaxKind.SimpleAssignmentExpression))
                 {
                     // Prop = value --> set
                     // value = Prop --> get
-                    return assignmentExpression.Left == topmostSyntax
-                        ? AccessorAccess.Set
-                        : AccessorAccess.Get;
+                    return assignmentExpression.Left == topmostSyntax ? AccessorAccess.Set : AccessorAccess.Get;
                 }
                 else
                 {
                     // Prop += value --> get/set
                     return AccessorAccess.Both;
                 }
             }
-
-            // nameof(Prop) --> get/set
-            if (node.IsInNameOfArgument(GetSemanticModel(node)))
+            else if (topmostSyntax.Parent is ArgumentSyntax argument && argument.IsAssignmentToTuple())
```

Comment:
this is for `(a,b) = someValue`, which is indeed Set.
How about `someValue = (a,b)` which should be a Get - could you add a test for it?